application/src/classes/states/track-loader.js

```diff
@@ -425,29 +425,24 @@ TrackLoaderState.prototype.updateCamera = function()
             averagePlayerPosition[0] += this.cars[i].x;
             averagePlayerPosition[1] += this.cars[i].y;
             carCount += 1;
-        }
 
-        squaredDistance = (
-            Math.pow(this.cars[i].x - nextMarker.x, 2) +
-            Math.pow(this.cars[i].y - nextMarker.y, 2)
-        );
+            squaredDistance = (
+                Math.pow(this.cars[i].x - nextMarker.x, 2) +
+                Math.pow(this.cars[i].y - nextMarker.y, 2)
+            );
 
-        if (squaredDistance < closestSquaredDistance) {
-            closestSquaredDistance = squaredDistance;
-            closestCar             = {
-                x : this.cars[i].x,
-                y : this.cars[i].y
-            };
+            if (squaredDistance < closestSquaredDistance) {
+                closestSquaredDistance = squaredDistance;
+                closestCar             = {
+                    x : this.cars[i].x,
+                    y : this.cars[i].y
+                };
+            }
         }
     }
 
-    if (carCount > 0) {
```

Comment:
I only ever saw that happen when switching states, because we weren't calling shutdown on the old state. You're right though, there should be a safeguard.